src/main/java/org/opensearch/flowframework/workflow/RegisterLocalModelStep.java

```diff
@@ -0,0 +1,196 @@
+        String modelGroupId = null;
+        String modelContentHashValue = null;
+        String modelType = null;
+        String embeddingDimension = null;
+        FrameworkType frameworkType = null;
+        String allConfig = null;
+        String url = null;
+
+        for (WorkflowData workflowData : data) {
+            Map<String, Object> content = workflowData.getContent();
+
+            for (Entry<String, Object> entry : content.entrySet()) {
+                switch (entry.getKey()) {
+                    case NAME_FIELD:
+                        modelName = (String) content.get(NAME_FIELD);
+                        break;
+                    case VERSION_FIELD:
+                        modelVersion = (String) content.get(VERSION_FIELD);
+                        break;
+                    case DESCRIPTION_FIELD:
+                        description = (String) content.get(DESCRIPTION_FIELD);
+                        break;
+                    case MODEL_FORMAT:
+                        modelFormat = MLModelFormat.from((String) content.get(MODEL_FORMAT));
+                        break;
+                    case MODEL_GROUP_ID:
+                        modelGroupId = (String) content.get(MODEL_GROUP_ID);
+                        break;
+                    case MODEL_TYPE:
+                        modelType = (String) content.get(MODEL_TYPE);
+                        break;
+                    case EMBEDDING_DIMENSION:
+                        embeddingDimension = (String) content.get(EMBEDDING_DIMENSION);
+                        break;
+                    case FRAMEWORK_TYPE:
+                        frameworkType = FrameworkType.from((String) content.get(FRAMEWORK_TYPE));
+                        break;
+                    case ALL_CONFIG:
+                        allConfig = (String) content.get(ALL_CONFIG);
+                        break;
+                    case MODEL_CONTENT_HASH_VALUE:
+                        modelContentHashValue = (String) content.get(MODEL_CONTENT_HASH_VALUE);
+                        break;
+                    case URL:
+                        url = (String) content.get(URL);
+                        break;
+                    default:
+                        break;
+
+                }
+            }
+        }
+
+        if (Stream.of(modelName, modelVersion, modelFormat, modelGroupId, embeddingDimension, frameworkType, modelContentHashValue, url)
+            .allMatch(x -> x != null)) {
+
+            // Create model configuration, assuming null pooling mode, null model max length, normalize results set to false
+            TextEmbeddingModelConfigBuilder modelConfigBuilder = TextEmbeddingModelConfig.builder()
+                .modelType(modelType)
+                .embeddingDimension(Integer.valueOf(embeddingDimension))
+                .frameworkType(frameworkType)
+                .poolingMode(null)
+                .modelMaxLength(null)
+                .normalizeResult(false);
```

Comment:
We are setting these values statically. Any reason?

Reply:
These are the default values for remote model registration, but looking through the [TextEmbeddingModelConfig](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/model/TextEmbeddingModelConfig.java) class again, I can remove these from our end